### DISCUSSION TITLE
Vehicles visit a similar number of stops

### DISCUSSION THREAD

**Discussion by TeodoraBogdan** [google/or-tools#2430] [Routing (and legacy CP) questions]
**What language and solver does this apply to?**
C++ / Routing,

**Describe the problem you are trying to solve.**

I have a problem with 50 stops and 3 vehicles and I want that each vehicle to similar (not equal) number of stops.
![Screenshot (79)](https://user-images.githubusercontent.com/52992360/110454379-d3818b00-80cf-11eb-930f-c20da2d2e783.png)

After the optimization I receive 3 routes, but 2 of them have only 3 stops. (I set the minimum number of stops visited to be 2 + the depot, to be sure that all the vehicles are used). In the red circle are the stops of one first route, in the blue circle are the stops of the second route and the third route has the rest of the stops.
![Screenshot (78)_LI](https://user-images.githubusercontent.com/52992360/110454503-fd3ab200-80cf-11eb-9f29-0886df3102e0.jpg)

**Describe the solution you'd like**

What I want is something like this:
![Screenshot (79)_LI](https://user-images.githubusercontent.com/52992360/110454692-383ce580-80d0-11eb-989c-660b9cf80c87.jpg)

I hope that there is a solution for this.

**Comment by Mizux** [comment]
Create a counter dimension and add some `SetCumulVarSoftLowerBound()`/`SetCumulVarSoftUpperBound()`

e.g. in Python
```python
 # Try to equaly distribute visit among the fleet
    routing.AddConstantDimension(
        1,
        manager.GetNumberOfNodes(),
        True,  # start cumul to zero
        "Counter")
    counter_dimension = routing.GetDimensionOrDie("Counter")
    num_visit = manager.GetNumberOfNodes() // manager.GetNumberOfVehicles()
    print(f'num of visit per vehicle: {num_visit}')

    for vehicle_id in range(manager.GetNumberOfVehicles()):
        index = routing.End(vehicle_id)
        counter_dimension.SetCumulVarSoftLowerBound(index, num_visit, 100_000)
        counter_dimension.SetCumulVarSoftUpperBound(index, num_visit + 1, 100_000)
```

**Reply by TeodoraBogdan** [reply]
Thank you for the response, but as I said above, I don't want them to have the exact number of stops; with your solution, if I have 50 stops and 3 vehicles, the routes will have 16, 17 and 17 stops.
If you look in the third photo the red route has 11 stops, the blue one has 21 and the yellow one has 18.

**Reply by Mizux** [reply]
So did you try to add a GlobalSpanCoefficient ?
What do you want to minimize ? the longest road ?
Why did you expect your solution ? Why it is better, or why it would improve the objective ?

BTW you should provide us a gist so we can reproduce it...

note: We don't see any depot in your image...
